solara/server/starlette.py

```diff
@@ -399,19 +399,17 @@ async def root(request: Request, fullpath: str = ""):
 """)
     if settings.oauth.private and not has_auth_support:
         raise RuntimeError("SOLARA_OAUTH_PRIVATE requires solara-enterprise")
-    root_path = settings.main.root_path or ""
-    if not settings.main.base_url:
+    root_path = settings.main.root_path
+    if not settings.main.origin:
         # Note:
         # starlette does not respect x-forwarded-host, and therefore
         # base_url and expected_origin below could be different
         # x-forwarded-host should only be considered if the same criteria in
         # uvicorn's ProxyHeadersMiddleware accepts x-forwarded-proto
-        settings.main.base_url = str(request.base_url)
-    # if not explicltly set,
-    configured_root_path = settings.main.root_path
+        settings.main.origin = request.base_url.scheme + "://" + request.base_url.netloc
     scope = request.scope
     root_path_asgi = scope.get("route_root_path", scope.get("root_path", ""))
-    if settings.main.root_path is None:
+    if settings.main.root_path == "":
```

Comment:
I think we want to keep this to None, to make sure we only do this when it is not explicitly configured.